shared/health/src/main/java/io/pravega/shared/health/DetailsProvider.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.health;
+
+import lombok.Getter;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * A details object encapsulates any health related, non-{@link Status} state that an operator may be interested in. A details
+ * object should provide information that helps gauge the well-being of the associated component.
+ */
+public class DetailsProvider {
```

Comment:
This class is not thread safe. Consider using `ConcurrentHashMap` or `Collections.synchronizedMap`.

Then mark your class as `@ThreadSafe`